__tests__/app.test.js

```diff
@@ -191,3 +191,104 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Inserts a new comment to the db and sends the comment back to the client", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toHaveProperty("comment_id");
+        expect(comment).toHaveProperty("votes");
+        expect(comment).toHaveProperty("created_at");
+        expect(comment).toHaveProperty("author");
+        expect(comment).toHaveProperty("body");
+        expect(comment).toHaveProperty("article_id");
+      });
+  });
+
+  test("400: Responds with an bad request error message when an non-existing username is provided", () => {
+    const newComment = {
+      username: "wrong_username",
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
+
+  test("400: Responds with an bad request error message when an no username is provided", () => {
+    const newComment = {
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
+
+  test("400: Responds with an bad request error message when an no comment is provided", () => {
+    const newComment = {
+      username: "butter_bridge",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
+
+  test("400: Responds with an bad request error message when the article doesn't exist", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "New comment",
+    };
+
+    return request(app)
+      .post("/api/articles/9999/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
```

Comment:
Similar to the above this is also a 404. This is because the id `9999` is valid but simply does not exist.

Have a read of the notes for more info on this as well:
https://l2c.northcoders.com/courses/sd-notes/back-end#sectionId=error-handling,step=intro